### PR TITLE
Update "Goto Command" to support following a .-delimited subcommand 

### PR DESCRIPTION
**Pull request by apbodnar**
Update CommandTreeNodeForCommand to generate a full node index for a
full SubCmdIdx.  Also adds sanity checks to the input logic that
disables the "OK" button and presents an error message upon invalid
input.

### CODE REVIEW

**Comment by apbodnar** [comment]
Ready for review

File: gapic/src/main/com/google/gapid/views/GotoCommand.java [hunk 1]
```diff
@@ -92,8 +92,12 @@ protected Control createCustomArea(Composite parent) {
...
       String[] strings = text.getText().split("\\.");
       value = new ArrayList<Long>();
-      for (String s : strings) {
-        value.add(Long.parseLong(s));
+      try {
+        for (String s : strings) {
+          value.add(Long.parseLong(s));
+        }
+      } catch ( NumberFormatException e) {
+        // TODO(apbodnar) Show some sort of warning/error
```
Comment (chrisforbes) [inline gapic/src/main/com/google/gapid/views/GotoCommand.java]: What's the right thing to do here?

**Comment by chrisforbes** [comment]
I'm happy with the backend changes. I think we should be a bit more tidy about the error handling in the UI.

File: gapis/resolve/command_tree.go [hunk 2]
```diff
@@ -75,19 +75,30 @@ func (t *commandTree) index(indices []uint64) (api.SpanItem, api.SubCmdIdx) {
...
-	for {
-		i := group.IndexOf(id)
-		out = append(out, i)
-		switch item := group.Index(i).(type) {
-		case api.CmdIDGroup:
-			group = item
-		default:
-			return out
+
+	for _, id := range idx {
```
Comment (AWoloszyn) [inline gapis/resolve/command_tree.go]: jump_label:
for _, id := range idx {
    for {
          ....
          default:
            break jump_label
    }
}

Saves turning it into a coroutine.

Comment (apbodnar) [reply gapis/resolve/command_tree.go]: Removed the anonymous function in a way that won't displease Dijkstra

File: gapic/src/main/com/google/gapid/views/GotoCommand.java (hunk 1 above)
Comment (pmuetschard) [reply gapic/src/main/com/google/gapid/views/GotoCommand.java]: There are two approaches we can take:
1) proactively parse the input whenever it changes, and disable the OK button and show a message if the input is invalid.
2) parse only when the user clicks OK, and don't close the dialog and show a message if the input is invalid. If you don't call `super.buttonPressed(..)`, the dialog will not close and stay around.

Personally, I prefer option 1.

**Comment by pmuetschard** [comment]
Please write a more detailed PR title and description.

File: gapic/src/main/com/google/gapid/views/GotoCommand.java (hunk 1 above)
Comment (chrisforbes) [reply gapic/src/main/com/google/gapid/views/GotoCommand.java]: Option 1 is much nicer :)

Comment (apbodnar) [reply gapic/src/main/com/google/gapid/views/GotoCommand.java]: Updated to disable the OK button and show a simple error message.  Though there's a good chance i'm doing something suboptimal with SWT

File: gapic/src/main/com/google/gapid/views/GotoCommand.java [hunk 3]
```diff
@@ -77,27 +82,35 @@ protected boolean isResizable() {
...
+      text = createTextbox(container, "");
+      text.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+      error = createLabel(container, "");
+      error.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+      text.addListener(SWT.Modify, evt -> {
+        String input = text.getText();
+        Button button = getButton(IDialogConstants.OK_ID);
+        error.setVisible(false);
+        error.setText("");
+        if (button != null && input.length() > 0) {
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/GotoCommand.java]: if the input is empty we should also disable the button.

File: gapic/src/main/com/google/gapid/views/GotoCommand.java [hunk 4]
```diff
@@ -77,27 +82,35 @@ protected boolean isResizable() {
...
-
       Composite container = createComposite(parent, new GridLayout(2, false));
       container.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
       createLabel(container, Messages.COMMAND_ID + ":");
-      spinner = createSpinner(container, current, 0, max);
-      spinner.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+      text = createTextbox(container, "");
+      text.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+      error = createLabel(container, "");
+      error.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/GotoCommand.java]: add:
`error.setForeground(getDisplay().getSystemColor(SWT.COLOR_DARK_RED));`

File: gapic/src/main/com/google/gapid/views/GotoCommand.java [hunk 5]
```diff
@@ -77,27 +82,35 @@ protected boolean isResizable() {
...
+          try {
+            String[] strings = input.split("\\.");
+            value = new ArrayList<Long>();
+            for (String s : strings) {
+              value.add(Long.parseLong(s));
+            }
+            button.setEnabled(true);
+          } catch (NumberFormatException e) {
+            error.setText("Invalid index: " + e.getMessage());
+            error.setVisible(true);
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/GotoCommand.java]: add:
`error.requestLayout();`

**Pull request merged**